python/packages/nisar/workflows/rubbersheet.py

```diff
@@ -22,9 +24,193 @@
+                                  key.startswith('layer')]
+
+                    # Get correlation peak path for the last offset layer since it has larger window size
+                    corr_peak_path = str(f'{off_prod_dir}/{layer_keys[-1]}/correlation_peak')
+                    dense_offsets_path =  str(f'{off_prod_dir}/{layer_keys[-1]}/dense_offsets')
+
+                # Sampling
+                az_sampled_indices = np.linspace(0,
+                                                 len(off_az_indices)-1,
+                                                 rubbersheet_params['polyfitting']['samples_along_azimuth'],
+                                                 dtype=int)
+                rg_sampled_indices = np.linspace(0,
+                                                 len(off_rg_indices)-1,
+                                                 rubbersheet_params['polyfitting']['samples_along_range'],
+                                                 dtype=int)
+
+                az_polyfit_indices = off_az_indices[az_sampled_indices]
+                rg_polyfit_indices = off_rg_indices[rg_sampled_indices]
+
+                lines, pixels = np.meshgrid(az_sampled_indices,
+                                            rg_sampled_indices,
+                                            indexing="ij")
+
+                corr_peak = _open_raster(corr_peak_path, 1)
+                az_offsets = _open_raster(dense_offsets_path, 1)
+                rg_offsets = _open_raster(dense_offsets_path, 2)
+
+                corr_peak = corr_peak[lines, pixels]
+                az_offsets = az_offsets[lines, pixels]
+                rg_offsets = rg_offsets[lines, pixels]
+
+                data = np.column_stack([
+                    np.arange(corr_peak.size),
+                    np.repeat(az_polyfit_indices, len(rg_polyfit_indices)),
+                    np.tile(rg_polyfit_indices, len(az_polyfit_indices)),
+                    az_offsets.ravel(), rg_offsets.ravel(),
+                    corr_peak.ravel()
+                ])
+
+                # polyfitting the range and azimuth offsets
+                results = offsets_polyfit.polyfit_offsets(
+                    data,degree=rubbersheet_params['polyfitting']['degree'],
+                    crit_value=rubbersheet_params['polyfitting']['critical_value'],
+                    max_iterations=len(data),
+                    minL=minL, maxL=maxL,
+                    minP=minP, maxP=maxP,
+                    prf=prf, rbw=rbw,
+                    abw=abw, rsr=rsr,)
+
+                print(f"Polyfitting Degree: {results["degree"]}")
+                print(f"Polyfitting CoefL: {results["coefL"]}")
+                print(f"Polyfitting CoefP: {results["coefP"]}")
+
+                info_channel.log(f"Polyfitting Degree: {results["degree"]}")
+                info_channel.log(f"Polyfitting CoefL: {results["coefL"]}")
+                info_channel.log(f"Polyfitting CoefP: {results["coefP"]}")
```

Comment:
```suggestion
                polyfitting_degree = results["degree"]
                polyfitting_coefL = results["coefL"]
                polyfitting_coefP = results["coefP"]

                print(f"Polyfitting Degree: {polyfitting_degree}")
                print(f"Polyfitting CoefL: {polyfitting_coefL}")
                print(f"Polyfitting CoefP: {polyfitting_coefP}")

                info_channel.log(f"Polyfitting Degree: {polyfitting_degree}")
                info_channel.log(f"Polyfitting CoefL: {polyfitting_coefL}")
                info_channel.log(f"Polyfitting CoefP: {polyfitting_coefP}")
```
Our Python Minimum Requirements Github Actions workflow fails due to how these strings are structured, presumably because you've embedded a second set of quotes into the format strings and older python strings don't like that.